### ISSUE TITLE
[search] There's an error in the logs when creating a site and editing a page for 3.0 and 3.1

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#2963]
## Describe the bug
There's an error in the logs when creating a site and editing a page for 3.0 and there's also an error in the logs when creating a site in 3.1 using Crafter Search with Solr

## To Reproduce
**For 3.0:**
1. Clone then build and deploy craftercms
2. Create a site using either website_editorial or headless_blog bp
3. Notice the errors in the logs

```
[ERROR] 2019-03-19 09:53:26,913 [http-nio-8080-exec-8] [controller.ExceptionHandlers] | Request /crafter-search/api/2/search/update failed with status 500
org.craftercms.search.exception.SearchException: [mysite2] Error executing update for mysite2:/site/website/articles/2016/12/top-books-for-young-women/index.xml
```
4. If you created a site using the website_editorial, edit the `Home` page and save your changes
5. Notice the errors in the logs after saving your changes to the home page

**For 3.1**
1. Clone then build and deploy craftercms
2. Start Solr when you start craftercms `./gradlew -Penv=authoring -PwithSolr=true start`
3. Create a site using the website_editorial bp or the headless_blog bp, or the headless_store bp and setup Crafter Search with Solr
4. Notice the errors in the logs 

## Expected behavior
There should be no errors when creating a site using the website_editorial or headless_blog bp, or when editing the home page of the website_editorial bp

## Logs
Error in the logs when creating a site in 3.0:
https://gist.github.com/alhambrav/78cac1a28287b75743600a703361a816

Error in the logs when editing home page on website_editorial in 3.0
https://gist.github.com/alhambrav/96dd9914d585d35ac3823f17d468930e

Error in the logs when creating a site in 3.1:
https://gist.github.com/alhambrav/1bef4cb44a1308a4490a49cdee53476c

### Version
For 3.0
Studio Version Number: 3.0.21-SNAPSHOT-dd243e
Build Number: dd243ebb01da86fb1ef695897b28bf09e6732b91
Build Date/Time: 03-19-2019 09:31:36 -0400

For 3.1
Studio Version Number: 3.1.0-SNAPSHOT-de0cd3
Build Number: de0cd3accb3b23bbeb90f9660559766a46298342
Build Date/Time: 03-19-2019 13:41:02 -0400

### OS
OS X

### Browser
Chrome browser

**Comment by dejan-brkic** [comment]
This is `search` issue. Studio completes saving successfully. Error stacktrace is generated when search index is updated

**Comment by alhambrav** [comment]
Verified fixed for 3.0

**Comment by alhambrav** [comment]
Verified fixed for 3.1
